buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ThirdPartyAuditTask.java

```diff
@@ -171,19 +172,30 @@ private void extractJars(FileCollection jars) {
         File jarExpandDir = getJarExpandDir();
         // We need to clean up to make sure old dependencies don't linger
         getProject().delete(jarExpandDir);
-        jars.forEach(jar ->
+
+        jars.forEach(jar -> {
+            FileTree jarFiles = getProject().zipTree(jar);
             getProject().copy(spec -> {
+                spec.from(jarFiles);
+                spec.into(jarExpandDir);
+                // exclude classes from multi release jars
+                spec.exclude("META-INF/versions/**");
+            });
+            // The order is important, we iterate here so we don't depend on the order in which Gradle executes the spec
+            IntStream.rangeClosed(
+                Integer.parseInt(JavaVersion.VERSION_1_9.getMajorVersion()),
```

Comment:
Shouldn't this be the compiler version? For example, while we don't currently, we could add java 10 or 11 specific classes.

Reply:
`targetCompatability` is being set to runtime, since we cycle trough the supported run-times in CI we will eventually test each. The goal is to extract everything from   9 ( the first to support multi release jars ) to the current runtime version, whith each extract overwriting the previously existing classes. This means that the result will be the exact classes the loader would pick when running on a particular java version. 

i will also add a comment in the code to clarify this.

Reply:
Ok, I see it now. Thanks for the explanation.